### PR TITLE
Update CMakeLists.txt with install support

### PR DESCRIPTION
**Pull request by pdimov**

### CODE REVIEW

**Comment by pdimov** [comment]
This is ready to merge, I think.

**Comment by pdimov** [comment]
I am going to merge this (sorry) because I want to use Atomic in the CI tests for tools/cmake. Atomic is the only compiled library that can have install support at the moment. The next contender, System, requires Predef to have install support (via Winapi), and it doesn't.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,26 +1,39 @@
 # Copyright 2018 Mike Dev
+# Copyright 2019 Peter Dimov
 # Distributed under the Boost Software License, Version 1.0.
 # See accompanying file LICENSE_1_0.txt or copy at https://www.boost.org/LICENSE_1_0.txt
 
-cmake_minimum_required(VERSION 3.5)
-project(BoostAtomic LANGUAGES CXX)
+cmake_minimum_required(VERSION 3.5...3.16)
```
Comment (Lastique) [inline CMakeLists.txt]: Does this add an upper limit for supported CMake version? Why is this limit needed? Does it break on newer versions?

Comment (pdimov) [reply CMakeLists.txt]: No. 3.5 by itself is equivalent to 3.5...3.5. The first value controls when to issue a hard error (on 3.4 and earlier.) The second value controls CMake behavior and policies. For compatibility, a 3.16 CMake will act like a 3.5 one when (VERSION 3.5) is used. (3.5...3.16) says we prefer 3.16 to act like 3.16.

**Pull request merged**